dart/constraint/ConstrainedGroup.h

```diff
@@ -107,7 +108,7 @@ class ConstrainedGroup
   std::vector<ConstraintBase*> mConstraints;
 
   ///
-  dynamics::Skeleton* mRootSkeleton;
+  std::shared_ptr<dynamics::Skeleton> mRootSkeleton;
```

Comment:
What would you think of a file named "Ptr.h", so that it's called "dart/dynamics/Ptr.h" when being included?